milli/src/update/index_documents/extract/mod.rs

```diff
@@ -43,18 +42,10 @@ pub(crate) fn data_from_obkv_documents(
     flattened_obkv_chunks: impl Iterator<Item = Result<grenad::Reader<BufReader<File>>>> + Send,
     indexer: GrenadParameters,
     lmdb_writer_sx: Sender<Result<TypedChunk>>,
-    searchable_fields: Option<HashSet<FieldId>>,
-    faceted_fields: HashSet<FieldId>,
     primary_key_id: FieldId,
     geo_fields_ids: Option<(FieldId, FieldId)>,
-    field_id_map: FieldsIdsMap,
-    stop_words: Option<fst::Set<Vec<u8>>>,
-    allowed_separators: Option<&[&str]>,
-    dictionary: Option<&[&str]>,
+    settings_diff: &Arc<InnerIndexSettingsDiff>,
```

Comment:
```suggestion
    settings_diff: Arc<InnerIndexSettingsDiff>,
```

```suggestion
    settings_diff: &InnerIndexSettingsDiff,
```

Sharing a reference to an `Arc` is an anti-pattern since its purpose is to replace a ref.
[Since `Arc<T>` implements `Deref<T>`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#impl-AsRef%3CT%3E-for-Arc%3CT,+A%3E) I believe you should be able to accept a simple reference here